Tools/i18n/msgfmt.py

```diff
@@ -213,21 +245,19 @@ def make(filename, outfile):
             sys.exit(1)
     # Add last entry
     if section == STR:
-        add(msgctxt, msgid, msgstr, fuzzy)
-
-    # Compute output
-    output = generate()
+        add(msgctxt, msgid, msgstr, fuzzy, messages)
 
+def writefile(outfile, output):
     try:
         with open(outfile,"wb") as f:
             f.write(output)
     except IOError as msg:
         print(msg, file=sys.stderr)
 
 
-def main():
+def main(argv):
```

Comment:
```suggestion
def main():
```
Why change this? sys.argv is more common in stdlib/tools from my experience.

Reply:
I thought that passing argv that way allowed simpler tests for the decoding of the command line. But as you are more experimented in CPython than I am, I shall follow your advice :-)

Reply:
That’s right but only for tests that import and call a main function, not when calling as a script 🙂